<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Holo-omics workbook | _main.knit</title>
  <meta name="description" content="" />
  <meta name="generator" content="bookdown 0.31 and GitBook 2.6.7" />

  <meta property="og:title" content="Holo-omics workbook | _main.knit" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Holo-omics workbook | _main.knit" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="holo-omics.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Holo-omics workbook</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#contents"><i class="fa fa-check"></i>Contents</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#protocols-exercises-and-tutorials"><i class="fa fa-check"></i>Protocols, exercises and tutorials</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#how-to-cite-this-work"><i class="fa fa-check"></i>How to cite this work</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#acknowledgement"><i class="fa fa-check"></i>Acknowledgement</a></li>
</ul></li>
<li class="part"><span><b>I INTRODUCTION</b></span></li>
<li class="chapter" data-level="1" data-path="holo-omics.html"><a href="holo-omics.html"><i class="fa fa-check"></i><b>1</b> Introduction to holo-omics</a>
<ul>
<li class="chapter" data-level="1.1" data-path="omic-layers.html"><a href="omic-layers.html"><i class="fa fa-check"></i><b>1.1</b> Omic layers</a></li>
<li class="chapter" data-level="1.2" data-path="host-genomics.html"><a href="host-genomics.html"><i class="fa fa-check"></i><b>1.2</b> Host genomics (HG)</a></li>
<li class="chapter" data-level="1.3" data-path="host-transcriptomics.html"><a href="host-transcriptomics.html"><i class="fa fa-check"></i><b>1.3</b> Host transcriptomics (HT)</a></li>
<li class="chapter" data-level="1.4" data-path="microbial-metagenomics.html"><a href="microbial-metagenomics.html"><i class="fa fa-check"></i><b>1.4</b> Microbial metagenomics (MG)</a></li>
<li class="chapter" data-level="1.5" data-path="microbial-metatranscriptomics.html"><a href="microbial-metatranscriptomics.html"><i class="fa fa-check"></i><b>1.5</b> Microbial metatranscriptomics (MT)</a></li>
<li class="chapter" data-level="1.6" data-path="host-proteomics.html"><a href="host-proteomics.html"><i class="fa fa-check"></i><b>1.6</b> Host proteomics (HP)</a></li>
<li class="chapter" data-level="1.7" data-path="microbial-metaproteomics.html"><a href="microbial-metaproteomics.html"><i class="fa fa-check"></i><b>1.7</b> Microbial metaproteomics (MP)</a></li>
<li class="chapter" data-level="1.8" data-path="meta-metabolomics.html"><a href="meta-metabolomics.html"><i class="fa fa-check"></i><b>1.8</b> (Meta)metabolomics (ME)</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="holofood.html"><a href="holofood.html"><i class="fa fa-check"></i><b>2</b> HoloFood</a>
<ul>
<li class="chapter" data-level="" data-path="holofood.html"><a href="holofood.html#methodological-point-of-view"><i class="fa fa-check"></i>Methodological point of view</a></li>
</ul></li>
<li class="part"><span><b>II LABORATORY PROCEDURES</b></span></li>
<li class="chapter" data-level="3" data-path="about-labwork.html"><a href="about-labwork.html"><i class="fa fa-check"></i><b>3</b> About labwork</a></li>
<li class="chapter" data-level="4" data-path="dna-rna-extraction.html"><a href="dna-rna-extraction.html"><i class="fa fa-check"></i><b>4</b> DNA/RNA extraction</a>
<ul>
<li class="chapter" data-level="" data-path="dna-rna-extraction.html"><a href="dna-rna-extraction.html#sample-preprocessing"><i class="fa fa-check"></i>Sample preprocessing</a></li>
<li class="chapter" data-level="" data-path="dna-rna-extraction.html"><a href="dna-rna-extraction.html#chemical-isolation"><i class="fa fa-check"></i>Chemical isolation</a></li>
<li class="chapter" data-level="" data-path="dna-rna-extraction.html"><a href="dna-rna-extraction.html#physicochemical-isolation"><i class="fa fa-check"></i>Physicochemical isolation</a></li>
<li class="chapter" data-level="" data-path="dna-rna-extraction.html"><a href="dna-rna-extraction.html#available-protocols"><i class="fa fa-check"></i>Available protocols</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="protein-metabolite-extraction.html"><a href="protein-metabolite-extraction.html"><i class="fa fa-check"></i><b>5</b> Protein/metabolite extraction</a></li>
<li class="chapter" data-level="6" data-path="sequencing-library-preparation.html"><a href="sequencing-library-preparation.html"><i class="fa fa-check"></i><b>6</b> Sequencing library preparation</a>
<ul>
<li class="chapter" data-level="6.1" data-path="subheading.html"><a href="subheading.html"><i class="fa fa-check"></i><b>6.1</b> Subheading</a></li>
</ul></li>
<li class="part"><span><b>III BIOINFORMATIC PROCEDURES</b></span></li>
<li class="chapter" data-level="7" data-path="about-bioinformatics.html"><a href="about-bioinformatics.html"><i class="fa fa-check"></i><b>7</b> About bioinformatics</a>
<ul>
<li class="chapter" data-level="7.1" data-path="prepare-your-shell-environment.html"><a href="prepare-your-shell-environment.html"><i class="fa fa-check"></i><b>7.1</b> Prepare your shell environment</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="sequencing-data-preprocessing.html"><a href="sequencing-data-preprocessing.html"><i class="fa fa-check"></i><b>8</b> Sequencing data preprocessing</a></li>
<li class="chapter" data-level="9" data-path="host-genomics-data-processing.html"><a href="host-genomics-data-processing.html"><i class="fa fa-check"></i><b>9</b> Host genomics (HG) data processing</a>
<ul>
<li class="chapter" data-level="9.1" data-path="host-reference-genome.html"><a href="host-reference-genome.html"><i class="fa fa-check"></i><b>9.1</b> Host reference genome</a></li>
<li class="chapter" data-level="9.2" data-path="host-genome-resequencing.html"><a href="host-genome-resequencing.html"><i class="fa fa-check"></i><b>9.2</b> Host genome resequencing</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="microbial-metagenomics-data-processing.html"><a href="microbial-metagenomics-data-processing.html"><i class="fa fa-check"></i><b>10</b> Microbial metagenomics (MG) data processing</a>
<ul>
<li class="chapter" data-level="10.1" data-path="reference-based.html"><a href="reference-based.html"><i class="fa fa-check"></i><b>10.1</b> Reference-based</a></li>
<li class="chapter" data-level="10.2" data-path="genome-resolved.html"><a href="genome-resolved.html"><i class="fa fa-check"></i><b>10.2</b> Genome-resolved</a></li>
<li class="chapter" data-level="10.3" data-path="genome-resolved-1.html"><a href="genome-resolved-1.html"><i class="fa fa-check"></i><b>10.3</b> Genome-resolved</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="host-transcriptomics-data-processing.html"><a href="host-transcriptomics-data-processing.html"><i class="fa fa-check"></i><b>11</b> Host transcriptomics (HT) data processing</a></li>
<li class="chapter" data-level="12" data-path="microbial-metatranscriptomics-data-processing.html"><a href="microbial-metatranscriptomics-data-processing.html"><i class="fa fa-check"></i><b>12</b> Microbial metatranscriptomics (HT) data processing</a></li>
<li class="chapter" data-level="13" data-path="host-proteomics-data-processing.html"><a href="host-proteomics-data-processing.html"><i class="fa fa-check"></i><b>13</b> Host proteomics (HP) data processing</a></li>
<li class="chapter" data-level="14" data-path="microbial-metaproteomics-data-processing.html"><a href="microbial-metaproteomics-data-processing.html"><i class="fa fa-check"></i><b>14</b> Microbial metaproteomics (MP) data processing</a></li>
<li class="part"><span><b>IV STATISTICAL PROCEDURES</b></span></li>
<li class="chapter" data-level="15" data-path="about-statistics.html"><a href="about-statistics.html"><i class="fa fa-check"></i><b>15</b> About statistics</a>
<ul>
<li class="chapter" data-level="15.1" data-path="prepare-your-r-environment.html"><a href="prepare-your-r-environment.html"><i class="fa fa-check"></i><b>15.1</b> Prepare your R environment</a>
<ul>
<li class="chapter" data-level="15.1.1" data-path="prepare-your-r-environment.html"><a href="prepare-your-r-environment.html#required-packages"><i class="fa fa-check"></i><b>15.1.1</b> Required packages</a></li>
<li class="chapter" data-level="15.1.2" data-path="prepare-your-r-environment.html"><a href="prepare-your-r-environment.html#package-installation"><i class="fa fa-check"></i><b>15.1.2</b> Package installation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="single-omic-analyses.html"><a href="single-omic-analyses.html"><i class="fa fa-check"></i><b>16</b> Single omic analyses</a></li>
<li class="chapter" data-level="17" data-path="data-transformation.html"><a href="data-transformation.html"><i class="fa fa-check"></i><b>17</b> Data transformation</a></li>
<li class="chapter" data-level="18" data-path="supervised-analysis.html"><a href="supervised-analysis.html"><i class="fa fa-check"></i><b>18</b> Supervised analysis</a></li>
<li class="chapter" data-level="19" data-path="unsupervised-exploration.html"><a href="unsupervised-exploration.html"><i class="fa fa-check"></i><b>19</b> Unsupervised exploration</a></li>
<li class="chapter" data-level="20" data-path="multi-omic-integration.html"><a href="multi-omic-integration.html"><i class="fa fa-check"></i><b>20</b> Multi-omic integration</a></li>
<li class="chapter" data-level="21" data-path="multi-staged-integration.html"><a href="multi-staged-integration.html"><i class="fa fa-check"></i><b>21</b> Multi-staged integration</a></li>
<li class="chapter" data-level="22" data-path="meta-dimensional-integration.html"><a href="meta-dimensional-integration.html"><i class="fa fa-check"></i><b>22</b> Meta-dimensional integration</a></li>
<li class="part"><span><b>V RESOURCES</b></span></li>
<li class="chapter" data-level="23" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>23</b> References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<!--bookdown:title:end-->
<!--bookdown:title:start-->
<div id="holo-omics-workbook" class="section level1 unnumbered hasAnchor">
<h1>Holo-omics workbook<a href="index.html#holo-omics-workbook" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The <strong>Holo-omics workbook</strong> is a compilation of methodological procedures to generate, analyse and integrate holo-omic data, i.e., multi-omic data jointly generated from hosts and associated microbial communities <span class="citation">[<a href="#ref-Nyholm-2020" role="doc-biblioref">1</a>, <a href="#ref-Limborg-2018" role="doc-biblioref">2</a>]</span>. This resource extends the contents of the article <strong>“A practical introduction to holo-omics”</strong>, which aims at guiding researchers to the main critical steps and decision points to perform holo-omic studies. While the article focuses on discussing pros and cons of using multiple available options, the aim of the workbook is to compile protocols and pipelines to be implemented by researchers. The <strong>Holo-omics Workbook</strong> is presented in two formats:</p>
<ul>
<li>Website (<a href="https://holo-omics.github.io/" class="uri">https://holo-omics.github.io/</a>)</li>
<li>PDF document (<a href="https://holo-omics.github.io/holo-omics_workbook.pdf" class="uri">https://holo-omics.github.io/holo-omics_workbook.pdf</a>)</li>
</ul>
<p>These resources are presented as two of the main final outputs of the H2020 project HoloFood. More information about this EU Innovation Action that ran between 2019 and 2023 can be found in the <a href="holofood.html#holofood">HoloFood section</a> in this workbook, the <a href="http://www.holofood.eu">HoloFood Website</a> and the <a href="https://cordis.europa.eu/project/id/817729">CORDIS website</a>.</p>
<div id="contents" class="section level3 unnumbered hasAnchor">
<h3>Contents<a href="index.html#contents" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><strong><a href="holo-omics.html#holo-omics">Introduction</a></strong>: general information about holo-omics, employed data types and study design considerations.</li>
<li><strong><a href="about-labwork.html#about-labwork">Laboratory procedures</a></strong>: methods and procedures for generating raw omic data of hosts and microbial communities.</li>
<li><strong><a href="about-bioinformatics.html#about-bioinformatics">Bioinformatic procedures</a></strong>: methods and procedures for processing raw omic data into quantitative datasets to be analysed through statistics.</li>
<li><strong><a href="about-statistics.html#about-statistics">Statistical procedures</a></strong>: methods and procedures for analysing and integrating holo-omic data.</li>
</ul>
</div>
<div id="protocols-exercises-and-tutorials" class="section level3 unnumbered hasAnchor">
<h3>Protocols, exercises and tutorials<a href="index.html#protocols-exercises-and-tutorials" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The workbook contains example data and bits of code (mostly shell and R) to reproduce data generation and analysis procedures. Code boxes look like the following:</p>
<pre><code>shao4d_perm &lt;- shao4d %&gt;%
  tax_transform(&quot;identity&quot;, rank = &quot;genus&quot;) %&gt;%
  dist_calc(&quot;aitchison&quot;) %&gt;%
  dist_permanova(
    variables = c(&quot;birth_mode&quot;, &quot;sex&quot;, &quot;number_reads&quot;),
    n_perms = 99, # you should use more permutations in your real analyses!
    n_processes = 1
  )
#&gt; Dropping samples with missings: 15
#&gt; 2022-11-24 01:15:20 - Starting PERMANOVA with 99 perms with 1 processes
#&gt; 2022-11-24 01:15:21 - Finished PERMANOVA

shao4d_perm %&gt;% perm_get()
#&gt; Permutation test for adonis under reduced model
#&gt; Marginal effects of terms
#&gt; Permutation: free
#&gt; Number of permutations: 99
#&gt;
#&gt; vegan::adonis2(formula = formula, data = metadata, permutations = n_perms, by = by, parallel = parall)
#&gt;               Df SumOfSqs      R2       F Pr(&gt;F)   
#&gt; birth_mode     1    10462 0.09055 29.3778   0.01 **
#&gt; sex            1      402 0.00348  1.1296   0.31   
#&gt; number_reads   1     1117 0.00967  3.1364   0.01 **
#&gt; Residual     287   102209 0.88462                  
#&gt; Total        290   115540 1.00000                  
#&gt; ---
#&gt; Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
</div>
<div id="how-to-cite-this-work" class="section level3 unnumbered hasAnchor">
<h3>How to cite this work<a href="index.html#how-to-cite-this-work" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Instructions to cite this work will be eventually added.</p>
</div>
<div id="acknowledgement" class="section level3 unnumbered hasAnchor">
<h3>Acknowledgement<a href="index.html#acknowledgement" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This project has received funding from the European Unionʼs Horizon 2020 research and innovation programme under grant agreement No 817729.</p>

</div>
</div>



<h3> References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body">
<div id="ref-Nyholm-2020" class="csl-entry">
1. Nyholm L, Koziol A, Marcos S, Botnen AB, Aizpurua O, Gopalakrishnan S, et al. Holo-omics: Integrated host-microbiota multi-omics for basic and applied biological research. iScience. 2020;23:101414.
</div>
<div id="ref-Limborg-2018" class="csl-entry">
2. Limborg MT, Alberdi A, Kodama M, Roggenbuck M, Kristiansen K, Gilbert MTP. Applied hologenomics: Feasibility and potential in aquaculture. Trends in Biotechnology. 2018;36:252–64.
</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="holo-omics.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "lunr",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true
},
"toolbar": {
"position": "fixed"
},
"info": true
});
});
</script>

</body>

</html>
